File: officefloor/polyglot/officepolyglot_scala/src/test/scala/net/officefloor/polyglot/scala/CollectionTypes.scala
package net.officefloor.polyglot.scala

class CollectionTypes(_list: java.util.List[Int], _set: java.util.Set[Long], _map: java.util.Map[String, Char], _collection: java.util.Collection[Byte]) {
  def getList() = _list
  def getSet() = _set
  def getMap() = _map
  def getCollection() = _collection
}

File: officefloor/polyglot/officepolyglot_test/src/main/java/net/officefloor/polyglot/test/CollectionTypes.java
package net.officefloor.polyglot.test;

import java.util.Collection;
import java.util.List;
import java.util.Map;
import java.util.Set;

import lombok.Value;

/**
 * {@link Collection} types.
 * 
 * @author Daniel Sagenschneider
 */
@Value
public class CollectionTypes {

	private final List<Integer> list;

	private final Set<Character> set;

	private final Map<String, JavaObject> map;

}